docs/design/fabric-v2+/fpc-registration.puml

```diff
@@ -7,266 +7,198 @@
+  return Credentials, Sealed_Credentials
 
-  create entity IAS order 22
-  Manager1 -> IAS ++: send quote
-  note right
-    use IAS credentials and url provided by config
-  end note
-  note over Manager1 #FFaa00
-    **MST**: From an attestation generalization perspective i think there should be just a single
-    call ""GetAttestationProof"" or alike rather than a separate place where you get quote
-    and another where you get the registration report which requires some re-organization of calls.
-    A few observations on this:
-    - we do not strictly have to call the enclave via ECC. ECCManager could also directly call
-      enclave.so. this would make this creation and key-management potentially easier?
-    - the quote generation is not driven from within the enclave as currently shown above but from
-      the outside!
-    I think from a high-level there should be following three high-level calls either in
-    ECC or ECCMAnager depending on above:
-    - EnclaveCreate (key-gen & imprint of context, i.e., current create/binding flows
-    - EnclaveProof: gets org endorsement-sig, creates report, then quote
-      and, if epid, gets IAS report. 
-    - EnclavePersist: returns sealed state (important for security that this is called only after
-      EnclaveProof [which we might want to enforce using some state-machine?] )
-  end note
-  return ATTESTATION
+  return Credentials, Sealed_Credentials
 
-  |||
+  E_Manager -> E_Manager : store Sealed_Credentials
 
-  Manager1 -> Manager1 : mark ECC_ID as installed
-  note right
-    Manager1 maintains a list of
-    locally "installed/created" enclaves
-  end note
-  return ATTESTATION, ECC_S_PK / error
+  return result /' Admin -> E_Manager '/
 
   deactivate Admin
-  |||
 
 end
 
-group register attestation
+group Org-Enclave binding/certification (to be implemented)
+  Admin -> E_Manager ++: certify(CC_Id, Cert_Parameters)
+  E_Manager -> ECC1 ++: getCSR(Cert_Parameters)
+  ECC1 -> Enclave1 ++: getCSR(Cert_Parameters)
+  return csr /' ECC1 -> Enclave1 '/
+  return csr /' E_Manager -> ECC1 '/
+  E_Manager -> E_Manager : generate peer certificate for enclave (certified by Org_1)
+  return certificate
+  note over Admin, E_Manager #FFaa00
+    -Org_1 takes "ownership" of this enclave,
+    -the certificate should be made available to the chaincode endorsement plugin,
+     so to include it in the responses (for eventual validation) 
+  end note
+end group
 
+group enclave registration
   activate Admin
+  Admin -> E_Manager ++: register
+  E_Manager -> ERCC1 ++: register Credentials
+  E_Manager -> ERCC2 : register Credentials
 
-  Admin -> ERCC1 ++: register (CC_NAME, CC_VER, MSP_ID_ORG_1, SIG_Org_1, ECC_S_PK, ECC_E_PK, ATTESTATION)
-  group verify and store
-
-    ERCC1 -> ERCC1 : check attestation signature & status
-    note left
-      - IAS verification key defined in ERCC definition
-      - status denotes the attestation result
+  ref over ERCC2
+    (same as ERCC1)
```

Comment:
I liked the previous reference to an actual group more. A pitty that `verify and store` group is not there anymore :/

Reply:
2b716a44f090dadd7906f75f2153becf2a9269a3